File: src/002.String/058.LengthOfLastWord/58.最后一个单词的长度.js
/*
 * @lc app=leetcode.cn id=58 lang=javascript
 *
 * [58] 最后一个单词的长度
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function (s) {
	const trimStr = s.trim();
	let lastLen = 0;

	for (let i = trimStr.length - 1; i >= 0; i--) {
		if (trimStr[i] == ' ') {
			break;
		}
		lastLen++;
	}

	return lastLen;
};
// @lc code=end

File: src/002.String/058.LengthOfLastWord/58.最后一个单词的长度.go
/*
 * @lc app=leetcode.cn id=58 lang=golang
 *
 * [58] 最后一个单词的长度
 */

// @lc code=start
func lengthOfLastWord(s string) int {
	index := len(s) - 1
	count := 0

	for s[index] == ' ' {
		index--
	}

	for index >= 0 && s[index] != ' ' {
		index--
		count++
	}
	return count
}

// @lc code=end

